crates/redeem/src/lib.rs

```diff
@@ -292,9 +307,14 @@ pub mod pallet {
         /// # Weight: `O(1)`
         #[pallet::weight(<T as Config>::WeightInfo::set_redeem_period())]
         #[transactional]
-        pub fn mint_tokens_for_reimbursed_redeem(origin: OriginFor<T>, redeem_id: H256) -> DispatchResultWithPostInfo {
-            let vault = ensure_signed(origin)?;
-            Self::_mint_tokens_for_reimbursed_redeem(vault, redeem_id)?;
+        pub fn mint_tokens_for_reimbursed_redeem(
+            origin: OriginFor<T>,
+            collateral_currency: CurrencyId<T>,
+            wrapped_currency: CurrencyId<T>,
```

Comment:
If we do decide to not flatten `VaultCurrenyPair`, perhaps we should use it here?